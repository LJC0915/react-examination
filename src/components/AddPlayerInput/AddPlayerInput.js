import React, { Component } from "react";
import classnames from "classnames";
import PropTypes from "prop-types";
import styles from "./AddPlayerInput.scss";

class AddPlayerInput extends Component {
  constructor(props, context) {
    super(props, context);
    this.state = {
      name: "",
      team: "",
      position: "PG"
    };
  }

  handleSubmit() {
    const { name, team, position } = this.state;
    this.props.addPlayer(name, team, position);
    this.setState({ name: "", team: "", position: "PG" });
  }

  render() {
    const { name, team, position } = this.state;
    return (
      <div className={styles.wrap}>
        <input
          type="text"
          className={classnames("form-control", styles.addPlayerInput)}
          placeholder="Name"
          value={name}
          onChange={e => this.setState({ name: e.target.value })}
        />
        <input
          type="text"
          className={classnames("form-control", styles.addPlayerInput)}
          placeholder="Team Name"
          value={team}
          onChange={e => this.setState({ team: e.target.value })}
        />
        <select
          value={position}
          className={classnames("form-control")}
          placeholder="Position"
          onChange={e => this.setState({ position: e.target.value })}
        >
          <option value="PG">PG</option>
          <option value="SG">SG</option>
          <option value="SF">SF</option>
          <option value="PF">PF</option>
          <option value="C">C</option>
        </select>
        <button
          className={` btn btn-primary `}
          onClick={() => this.handleSubmit()}
        >
          Add Player
        </button>
      </div>
    );
  }
}

AddPlayerInput.propTypes = {
  addPlayer: PropTypes.func.isRequired
};

export default AddPlayerInput;
