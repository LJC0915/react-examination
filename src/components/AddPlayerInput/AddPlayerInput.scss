:local(.addPlayerInput) {
  border-radius: 0;
  border-color: #abaaaa;
}

:local(.wrap) {
  display: flex;
  justify-content: space-between;
  background: #fff;
  padding: 10px 0;
  width: 100%;
  > * {
    width: 25%;
    margin: 0 5px;
  }
}
