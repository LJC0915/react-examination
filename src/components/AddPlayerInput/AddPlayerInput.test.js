import React from "react";
import AddPlayerInput from "./AddPlayerInput";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

describe("AddPlayerInput spec", () => {
  it("render AddPlayerInput correctly", () => {
    const mockAddPlayer = jest.fn();
    const wrapper = renderer
      .create(<AddPlayerInput addPlayer={mockAddPlayer} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger addPlayer correctly", () => {
    const mockAddPlayer = jest.fn();
    const wrapper = mount(<AddPlayerInput addPlayer={mockAddPlayer} />);
    const button = wrapper.find("button").at(0);
    button.simulate("click");
    expect(mockAddPlayer).toBeCalled();
  });
});
