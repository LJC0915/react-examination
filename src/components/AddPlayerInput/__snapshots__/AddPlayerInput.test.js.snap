// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPlayerInput spec render AddPlayerInput correctly 1`] = `
<div>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Team Name"
    type="text"
    value=""
  />
  <select
    className="form-control"
    onChange={[Function]}
    placeholder="Position"
    value="PG"
  >
    <option
      value="PG"
    >
      PG
    </option>
    <option
      value="SG"
    >
      SG
    </option>
    <option
      value="SF"
    >
      SF
    </option>
    <option
      value="PF"
    >
      PF
    </option>
    <option
      value="C"
    >
      C
    </option>
  </select>
  <button
    className=" btn btn-primary "
    onClick={[Function]}
  >
    Add Player
  </button>
</div>
`;
